legacy/src/main/java/org/opensearch/sql/legacy/query/OpenSearchActionFactory.java

```diff
@@ -116,7 +126,7 @@ public static QueryAction create(Client client, QueryActionRequest request)
         } else {
           sqlExpr.accept(new TermFieldRewriter());
           // migrate aggregation to query planner framework.
-          if (shouldMigrateToQueryPlan(sqlExpr, request.getFormat())) {
+          if (!bypassMigrateToQueryPlan && shouldMigrateToQueryPlan(sqlExpr)) {
```

Comment:
Previously some test cases which uses Format.JSON to bypasses this check and gets QueryAction for SELECT (either AggregationQueryAction or DefaultQueryAction). After removing Format.JSON, those test case wouldn't work without this flag.